### PR TITLE
classifier-comparison: first cut at classifier framework

### PR DESCRIPTION
**Pull request by jeffkaufman**
Example:

```
$ ./evaluate.py kraken-assigments kraken-hits p2ra no-human-hv-bowtie
                            totpos  truepos  trueneg  falsepos  falseneg
kraken-assigments              581        6      180         0        41
kraken-hits                    852        8      180         0        39
p2ra                           577        6      180         0        41
no-human-hv-bowtie            2391       36      158        22        11
no-bact-no-human-hv-bowtie    1968       36      175         5        11
```

See https://docs.google.com/document/d/1_xApUNo1TrdCZBcIp2YzqkOrOmLsrC03RMUJ96-Tf4E/edit

### CODE REVIEW

File: classifier-comparision/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        for read_id, cdata in load_data(sample):
+            read_notes = {}
+            if read_id in gt_read_ids:
+                read_notes["hv"] = read_id in ground_truth["yes"]
+
+            for classifier_name, classifier, metric in zip(
+                    classifier_names, classifiers, metrics):
+                out = classifier.classify(sample, read_id, cdata)
+                if type(out) == type(True):
+                    result = out
```
Comment (simonleandergrimm) [inline classifier-comparision/evaluate.py:123]: I don't fully understand how classifier_notes get's handled here, it would be helpful to talk through this.

**Review by simonleandergrimm — APPROVED** [review]
I didn't see any mistakes, though I found evaluate.py hard to parse. I left one specific question in that script that I'd be happy to understand before merging.

File: classifier-comparision/evaluate.py (hunk 1 above)
Comment (jeffkaufman) [reply classifier-comparision/evaluate.py:123]: We talked through this in person: notes are debug information that a classifier optionally generates
